doc/reference/misc/timeutil.rst

```diff
@@ -0,0 +1,289 @@
+TAI and UT varies over time. There are several variants of UT, with `UTC
+<timeutil-UTC>`_ being the most common.
+
+UT times are independent of location. UT is the basis for Standard Time
+(or "local time") which is the time at a particular location. Standard
+time has a fixed offset from UT at any given instant, primarily
+influenced by longitude, but the offset may be adjusted ("daylight
+saving time") to align standard time to the local solar time. In a sense
+local time is "more discontinuous" than UT.
+
+`POSIX Time <timeutil-RFC8536s2>`_ is a time scale that counts seconds since
+the "POSIX epoch" at 1970-01-01T00:00:00Z (i.e. the start of 1970
+UTC). `UNIX Time <timeutil-RFC8536s2>`_ is an extension of POSIX time using
+negative values to represent times before the POSIX epoch. Both of these
+scales assume that every day has exactly 86400 seconds. In normal use
+instants in these scales correspond to times in the UTC scale, so they
+inherit the discontinuity.
+
+The continuous analogue is `UNIX Leap Time <timeutil-RFC8536s2>`_ which
+is UNIX time plus all leap-second corrections added after the POSIX
+epoch (when TAI-UTC was 8 s).
+
+Example of Time Scale Differences
+---------------------------------
+
+A positive leap second was introduced at the end of 2016, increasing the
+difference between TAI and UTC from 36 seconds to 37 seconds. There was
+no leap second introduced at the end of 1999, when the difference
+between TAI and UTC was only 32 seconds. The following table shows
+relevant civil and epoch times in several scales:
+
+==================== ========== =================== ======= ==============
+UTC Date             UNIX time  TAI Date            TAI-UTC UNIX Leap Time
+==================== ========== =================== ======= ==============
+1970-01-01T00:00:00Z 0          1970-01-01T00:00:08 +8      0
+1999-12-31T23:59:28Z 946684768  2000-01-01T00:00:00 +32     946684792
+1999-12-31T23:59:59Z 946684799  2000-01-01T00:00:31 +32     946684823
+2000-01-01T00:00:00Z 946684800  2000-01-01T00:00:32 +32     946684824
+2016-12-31T23:59:59Z 1483228799 2017-01-01T00:00:35 +36     1483228827
+2016-12-31T23:59:60Z undefined  2017-01-01T00:00:36 +36     1483228828
+2017-01-01T00:00:00Z 1483228800 2017-01-01T00:00:37 +37     1483228829
+==================== ========== =================== ======= ==============
+
+Functional Requirements
+-----------------------
+
+The Zephyr tick counter has no concept of leap seconds or standard time
+offsets and is a continuous time scale. However it can be relatively
+inaccurate, with drifts as much as three minutes per hour (assuming an
+RC timer with 5% tolerance).
+
+There are two stages required to support conversion between Zephyr time
+and common human time scales:
+
+* Translation between the continuous but inaccurate Zephyr time scale
+  and an accurate external stable time scale;
+* Translation between the stable time scale and the (possibly
+  discontinuous) civil time scale.
```

Comment:
Do you mean that the civil time scale would be non-monotonic? That's how I interpreted your definition of the term, that you described [[here](https://github.com/zephyrproject-rtos/zephyr/issues/30590#issuecomment-744427593)]. 
Also, the stable time scale would/should be continuous(/monotonic) then?

Reply:
monotonic is well-defined: each value observed is no less than the previous value.

continuous follows the math definition: no abrupt changes in value.  For time it's generally saying it's linear.

TAI is a timescale that is monotonic and continuous.  Each time mark (representation of an instant) is totally ordered and maps one-to-one to an instant in time.

UTC is a timescale that is monotonic and discontinuous.  Each time mark is totally ordered and increasing, but some may be skipped, e.g. 2021-12-31T23:59:59 would not exist if a negative leap second applied at the end of this year.  (Or 2021-12-31T23:59:59 might be followed by 2021-12-31T23:59:60 before 2022-01-01T00:00:00 if there was a positive leap second.)

European Summer Time and other daylight-aware time scales are civil timescales that are not monotonic, and therefore discontinuous.  "civil" because it's a time scale legally defined by civil authorities, like local governments.  Not monotonic because time goes backward by 1 hour in fall, causing the same time mark to apply to two instants at a one hour interval.

So civil timescales can be non-monotonic, but the time scales used by this API must be monotonic and linear, so any any discontinuities must be addressed by external adjustments, just would be if somebody wanted time on a device to be in the local timezone-aware timescale.

Reply:
Thanks, that clarifies it a lot. If you think it would fit in the documentation, I think this information could be very useful in there as well.